canary/consumer-java/src/main/java/com/amazon/kinesis/video/canary/consumer/WebrtcStorageCanaryConsumer.java

```diff
@@ -0,0 +1,212 @@
+import com.amazonaws.services.cloudwatch.model.PutMetricDataRequest;
+import com.amazonaws.services.cloudwatch.model.StandardUnit;
+import com.amazonaws.services.kinesisvideo.model.StartSelector;
+import com.amazonaws.services.kinesisvideo.model.StartSelectorType;
+import com.amazonaws.kinesisvideo.parser.utilities.FrameVisitor;
+import com.amazonaws.kinesisvideo.parser.examples.GetMediaWorker;
+
+import lombok.extern.log4j.Log4j2;
+
+/*
+ * Canary for WebRTC with Storage thro Media Server
+ * 
+ * For longRun-configured jobs, this Canary will emit FragmentContinuity metrics by continuously
+ * checking for any newly ingested fragments for the given stream. The fragment list is checked
+ * for new fragments every "fragmentDuration * 2" The fragmentDuration is set by the encoder in
+ * configuration in Media Server.
+ * 
+ * For periodic-configured jobs, this Canary will emit TimeToFirstFragment metrics by continuously
+ * checking for consumable media from the specified stream via GetMedia calls.
+ */
+
+@Log4j2
+public class WebrtcStorageCanaryConsumer {
+    protected static final Date mCanaryStartTime = new Date();
+    protected static final String mStreamName = System.getenv("CANARY_STREAM_NAME"); 
+
+    private static final String mCanaryLabel = System.getenv("CANARY_LABEL");
+    private static final String mRegion = System.getenv("AWS_DEFAULT_REGION");
+
+    private static EnvironmentVariableCredentialsProvider mCredentialsProvider;
+    private static AmazonKinesisVideo mAmazonKinesisVideo;
+    private static AmazonCloudWatch mCwClient;
+
+    private static void calculateFragmentContinuityMetric(CanaryFragmentList fragmentList) {
+        try {
+            final GetDataEndpointRequest dataEndpointRequest = new GetDataEndpointRequest()
+                    .withAPIName(APIName.LIST_FRAGMENTS).withStreamName(mStreamName);
+            final String listFragmentsEndpoint = mAmazonKinesisVideo.getDataEndpoint(dataEndpointRequest)
+                    .getDataEndpoint();
+
+            TimestampRange timestampRange = new TimestampRange();
+            timestampRange.setStartTimestamp(mCanaryStartTime);
+            timestampRange.setEndTimestamp(new Date());
+
+            FragmentSelector fragmentSelector = new FragmentSelector();
+            fragmentSelector.setFragmentSelectorType("SERVER_TIMESTAMP");
+            fragmentSelector.setTimestampRange(timestampRange);
+
+            Boolean newFragmentReceived = false;
+
+            final FutureTask<List<CanaryFragment>> futureTask = new FutureTask<>(
+                    new CanaryListFragmentWorker(mStreamName, mCredentialsProvider, listFragmentsEndpoint,
+                            Regions.fromName(mRegion), fragmentSelector));
+            Thread thread = new Thread(futureTask);
+            thread.start();
+
+            List<CanaryFragment> newFragmentList = futureTask.get();
+            if (newFragmentList.size() > fragmentList.getFragmentList().size()) {
```

Comment:
Curious to know if it's possible that 2 fragments expired, but then 1 new fragment came in, this check wouldn't go through. It seems that we assume that the stream media expiration is long enough, but may want to leave a comment about this edge case

Reply:
Yes, good point, adding a comment.